src/bucket.ts

```diff
@@ -337,6 +337,115 @@ export interface BucketExistsCallback {
   (err: null, exists: boolean);
 }
 
+/**
+ * @typedef {object} [GetBucketRequest] Configuration options for Bucket#get()
+ * @property {boolean} [autoCreate] Automatically create the object if
+ *     it does not exist. Default: `false`
+ * @property {string} [userProject] The ID of the project which will be
+ *     billed for the request.
+ */
+export interface GetBucketRequest extends GetConfig {
+  userProject?: string;
+}
+
+/**
+ * @typedef {array} GetBucketResponse
+ * @property {Bucket} 0 The {@link Bucket}.
+ * @property {object} 1 The full API response.
+ */
+export type GetBucketResponse = [Bucket, object];
```

Comment:
Why `object`?  Can we just use `request.Response` here since that's what we're actually passing?